.buildkite/scripts/backport_branch.sh

```diff
@@ -0,0 +1,195 @@
+    git checkout $SOURCE_BRANCH -- $BUILDKITE_FOLDER_PATH
+    echo "Copying $JENKINS_FOLDER_PATH from $SOURCE_BRANCH..."
+    git checkout $SOURCE_BRANCH -- $JENKINS_FOLDER_PATH
+  else
+    git checkout $BACKPORT_BRANCH_NAME
+    echo "Copying $BUILDKITE_FOLDER_PATH from $SOURCE_BRANCH..."
+    git checkout $SOURCE_BRANCH -- $BUILDKITE_FOLDER_PATH
+    echo "Romoveing $JENKINS_FOLDER_PATH from $BACKPORT_BRANCH_NAME..."
+    rm -rf "$JENKINS_FOLDER_PATH"
+  fi
+
+  if [ "${REMOVE_OTHER_PACKAGES}" == "true" ]; then
+    echo "Removing all packages from $PACKAGES_FOLDER_PATH folder"
+    removeOtherPackages "$PACKAGES_FOLDER_PATH"
+    ls -la $PACKAGES_FOLDER_PATH
+  fi
+
+  echo "Setting up git environment..."
+  git config --global user.name "${GITHUB_USERNAME_SECRET}"
+  git config --global user.email "${GITHUB_EMAIL_SECRET}"
+
+  if [ "$DRY_RUN" == "true" ];then
+    echo "DRY_RUN mode, nothing will be pushed."
+    git diff $SOURCE_BRANCH...$BACKPORT_BRANCH_NAME
+  else
+    echo "Commiting and pushing..."
+    git add .
+    git commit -m "Add $BUILDKITE_FOLDER_PATH and $JENKINS_FOLDER_PATH to backport branch: $BACKPORT_BRANCH_NAME from the $SOURCE_BRANCH branch"
+    git push origin $BACKPORT_BRANCH_NAME
+  fi
+}
+
+if ! [[ $PACKAGE_VERSION =~ ^[0-9]+(\.[0-9]+){2}(\-.*)?$ ]]; then
+  buildkite-agent annotate "The entered package version ${PACKAGE_VERSION} doesn't match the pattern" --style "error"
+  exit 1
+fi
+
+add_bin_path
+
+with_yq
+
+echo "Check if the package is published"
+if ! isPackagePublished "$FULL_ZIP_PACKAGE_NAME"; then
+  buildkite-agent annotate "The package version: **${PACKAGE_NAME}-${PACKAGE_VERSION}** hasn't been published yet." --style "error"
+  exit 1
+fi
+
+echo "Check if the base commit exists."
+if [ ! -z "$BASE_COMMIT" ]; then
+  if ! commitExists "$BASE_COMMIT" "$SOURCE_BRANCH"; then
+    buildkite-agent annotate "The entered commit hasn't found in the **${SOURCE_BRANCH}** branch" --style "error"
+    exit 1
+  fi
+fi
+
+echo "Check if the backport-branch exists"
+MSG=""
+if ! branchExist "$BACKPORT_BRANCH_NAME"; then
+  echo "Check the entered version and PACKAGE_VERSION are equal"
+  version="$(cat packages/${PACKAGE_NAME}/manifest.yml | yq -r .version)"
+  if [[ "${version}" != "${PACKAGE_VERSION}" ]]; then
+    buildkite-agent annotate "Unexpected version found in packages/${PACKAGE_NAME}/manifest.yml" --style "error"
+    exit 1
+  fi
+
+  echo "Check that this changeset is the one creating the version $PACKAGE_NAME"
+  if ! git show -p ${BASE_COMMIT} packages/${PACKAGE_NAME}/manifest.yml | grep -E "^\+version: ${PACKAGE_VERSION}" ; then
+    buildkite-agent annotate "This changeset does not creates the version ${PACKAGE_VERSION}" --style "error"
+    exit 1
+  fi
+  createLocalBackportBranch "$BACKPORT_BRANCH_NAME" "$BASE_COMMIT"
+  MSG="The backport branch: **$BACKPORT_BRANCH_NAME** has been created."
+else
+  MSG="The backport branch: **$BACKPORT_BRANCH_NAME** has been updated."
```

Comment:
Probably this `else`  block could be removed, since the function `branchExist` already shows a message about it.

```suggestion
```

If not, I would suggest to change this message since the branch has not been updated at this point.

```suggestion
else
  MSG="The backport branch: **$BACKPORT_BRANCH_NAME** is already created."
```

Reply:
fixed MSG
I'm not sure about removing the `else `block.
this block prepares the message for BK annotation. the `branchExist `just shows the echo output to console.

Reply:
@mrodm , WDYT?

Reply:
Ok let's keep the message then.

In any case, we were discussing @jsoriano and me and if the branch is already created, it would be better not updating the contents of the branch. If the branch is already created, let's finish the script here without changing the branch.

All the current backport branches have been already updated manually, so it should not be needed to do that. Let's just update the contents when the branch is created.